tests/lists.test.ts

```diff
@@ -59,6 +59,17 @@ describe('List ([a])', () => {
       .otherwise(() => 'something else');
   });
 
+  it('issue #271: P.array should support readonly arrays as its input', () => {
+    type Input = string | Date | readonly string[];
+    const input = ['a', 'b', 'c'] as Input;
+
+    const output = match(input)
+      .with(P.array(P.string), (value) => 2)
+      .with(P.string, (value) => 1)
+      .with(P.instanceOf(Date), (value) => 3)
+      .exhaustive();
+  });
```

Comment:
Could you move this unit test to the `exhaustive-match.test.ts` file?

I think this is technically a bug of exhaustiveness checking, since the problem is that without this change, the match expression isn't considered exhaustive.

Reply:
@gvergnaud Sure! I moved it in this commit [980e4d0](https://github.com/gvergnaud/ts-pattern/pull/284/commits/980e4d085e0e55bb7e3d44f715f5fa9f63e5e66b)